Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -581,6 +581,24 @@ theorem ppow_succ' {f : ArithmeticFunction R} {k : ℕ} {kpos : 0 < k} :
 
 end Pmul
 
+section Pdiv
+
+/-- This is the pointwise division of `ArithmeticFunction`s. -/
+def pdiv [GroupWithZero R] (f g : ArithmeticFunction R) : ArithmeticFunction R :=
+  ⟨fun n => f n / g n, by simp only [map_zero, ne_eq, not_true, div_zero]⟩
+
+@[simp]
+theorem pdiv_apply [GroupWithZero R] (f g : ArithmeticFunction R) (n : ℕ) :
+    pdiv f g n = f n / g n := rfl
+
+@[simp]
+theorem pdiv_zeta [DivisionSemiring R] (f : ArithmeticFunction R) :
```

Comment:
Why does this need division semiring if the others were only group with zero

Reply:
Ah I see thanks, zeta is defined on Nat. I'd argue that this should probably be generalized to an arithmetic function valued in anything with a zero and a one (i.e. MonoidWithZero), and then your lemma should hold there, but making that generalization is a problem for a different PR.

Reply:
And yes we don't currently have a good way to do this. At one point we had a linter that tried to make these generalizations automatically for you, but it needs to be reconsidered and revamped for lean 4.

Reply:
I agree: probably worth generalising `zeta` but work for a separate PR.

However I think it is worth adding a comment above `pdiv_zeta` explaining that the specialisation of `zeta` is the reason for the `DivisionSemiring` class here.